RFCs/0000-pdk-template-v2.md

```diff
@@ -0,0 +1,296 @@
+DRAFT ONLY
+
+# PDK Templating Version 2.0
+
+## Summary
+
+> One paragraph explanation of the feature.
+
+The PDK templating engine originally started from [modulesync](https://github.com/voxpupuli/modulesync) which had a fairly simple use case: To generate the configuration files for the Puppet Labs supported modules.
+However since then the PDK use case has greatly expanded in scope. Catering for many CI services, creating files on convert, as well, as keeping files up to date.
+
+The templating system needs to be reviewed, and the architecture changed to support the ever growing need for end user scenarios.
+
+## Background & Assumptions
+
+> What is the context that someone needs to understand the goals of this proposal? This information should be objective
+> facts (i.e. not subjective opinions) about the current state of things.
+
+Evidence that the current templating system is in need of attention
+
+* The number of combinations of settings is continually growing. This is evident in the number of commits, and the growth of the pdk-templates repository
```

Comment:
Hrmm...yes. 

Would it be reasonable to say that the declining commit count, and declining number of average lines per commit would be due to it being too complex? That is, a lack of activity is partial result of the increasing complexity?

Reply:
I would say that the "The number of combinations of settings is continually growing" is easy to understand, that basically happens every time a new setting is added. It could probably be rephrased but the way I understood this was that with the current structure, the base template structure can only gain new complexity, there really isn't a mechanism for us to refactor things and reduce complexity due to the lack of flexibility in the template system itself.